src/main/java/com/hubspot/imap/utils/parsers/string/AtomOrStringParser.java

```diff
@@ -76,4 +81,52 @@ private void append(AppendableCharSequence seq, char c) {
     seq.append(c);
   }
 
+  /**
+   * See https://tools.ietf.org/html/rfc3501.html#section-4.3
+   *
+   * String literals have the form
+   *
+   * {10}
+   * abcdefghij
+   *
+   * Where {10} represents the length of the string literal. The literal
+   * begins after any CLRF characters following the '}' character.
+   */
+  private String parseLiteral(ByteBuf buffer, AppendableCharSequence seq) {
+    int length = 0;
+
+    char c = (char) buffer.readUnsignedByte();
+    while (c != '}') {
+      length = (length * 10) + Character.getNumericValue(c);
```

Comment:
Should we check that c is in fact numeric here? What does `Character.getNumericValue(c)` do when it is passed a non-numeric character?